### PR TITLE
feat(form-plugin): implement `propertyPath` parameter in the `UpdateFormValue`

### PR DESCRIPTION
**Pull request by arturovt**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #910 #260 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by arturovt** [comment]
@splincode could you review?

**Comment by arturovt** [comment]
Guys, I left `propertyPath`. 

P.S. as we're discussing it's not a problem to change the name of this property!

File: docs/plugins/form.md [hunk 1]
```diff
@@ -127,9 +127,90 @@ this.store.dispatch(
 The form plugin comes with the following `actions` out of the box:
 
 - `UpdateFormStatus({ status, path })` - Update the form status
-- `UpdateFormValue({ value, path })` - Update the form value
+- `UpdateFormValue({ value, path, propertyPath })` - Update the form value
```
Comment (markwhitfeld) [inline docs/plugins/form.md]:
```suggestion
- `UpdateFormValue({ value, path, propertyPath? })` - Update the form value (or optionally an inner property value)
```

File: docs/plugins/form.md [hunk 2]
```diff
@@ -127,9 +127,90 @@ this.store.dispatch(
...
+    ])
+  });
+
+  onSubmit() {
+    //
+  }
+}
+```
+
+Let's look at the component above again. Assume we want to update the first control's `name`. The code would look as follows:
```
Comment (markwhitfeld) [inline docs/plugins/form.md]:
```suggestion
Let's look at the component above again. Assume we want to update the name of the first author in our form, from anywhere in our application. The code would look as follows:
```

File: docs/plugins/form.md [hunk 3]
```diff
@@ -127,9 +127,90 @@ this.store.dispatch(
...
 - `UpdateFormStatus({ status, path })` - Update the form status
-- `UpdateFormValue({ value, path })` - Update the form value
+- `UpdateFormValue({ value, path, propertyPath })` - Update the form value
 - `UpdateFormDirty({ dirty, path })` - Update the form dirty status
 - `SetFormDisabled(path)` - Set the form to disabled
 - `SetFormEnabled(path)` - Set the form to enabled
 - `SetFormDirty(path)` - Set the form to dirty (shortcut for `UpdateFormDirty`)
 - `SetFormPristine(path)` - Set the form to pristine (shortcut for `UpdateFormDirty`)
+
+### Updating form properties separately
```
Comment (markwhitfeld) [inline docs/plugins/form.md]:
```suggestion
### Updating Form Properties Separately
```

File: packages/form-plugin/tests/update-form-properties.spec.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    const store = getStore();
+    const fixture = createComponent();
+    fixture.detectChanges();
+
+    // Act
+    store.dispatch(
+      new UpdateFormValue({
+        path,
+        value: {
+          name: 'Mark Whitfield'
```
Comment (markwhitfeld) [inline packages/form-plugin/tests/update-form-properties.spec.ts]: ;-)
```suggestion
          name: 'Mark Whitfeld'
```

File: packages/form-plugin/tests/update-form-properties.spec.ts [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+      })
+    );
+
+    const model = store.selectSnapshot(NovelsState.model);
+    const authors = fixture.debugElement.queryAll(By.css('.authors'));
+    const inputs = authors.map(author => author.query(By.css('input.author-name')));
+    const values = inputs.map(input => input.nativeElement.value);
+
+    // Assert
+    expect(values).toEqual(['Mark Whitfield', 'Artur', 'Max']);
```
Comment (markwhitfeld) [inline packages/form-plugin/tests/update-form-properties.spec.ts]:
```suggestion
    expect(values).toEqual(['Mark Whitfeld', 'Artur', 'Max']);
```

File: packages/form-plugin/tests/update-form-properties.spec.ts [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    expect(model.authors).toEqual([
+      {
+        name: 'Mark Whitfield'
```
Comment (markwhitfeld) [inline packages/form-plugin/tests/update-form-properties.spec.ts]:
```suggestion
        name: 'Mark Whitfeld'
```

**Comment by markwhitfeld** [comment]
Will we hold off on merging this until we have fixed more bugs on master?

**Comment by arturovt** [comment]
@splincode could you finish your HMR PR? We don't have any bug fixes, so we could publish 3.5.1. Other PRs are feature PRs. Deal? 😄

**Comment by arturovt** [comment]
@markwhitfeld would you approve?

File: docs/plugins/form.md [hunk 7]
```diff
@@ -127,9 +127,88 @@ this.store.dispatch(
 The form plugin comes with the following `actions` out of the box:
 
 - `UpdateFormStatus({ status, path })` - Update the form status
-- `UpdateFormValue({ value, path })` - Update the form value
+- `UpdateFormValue({ value, path, propertyPath? })` - Update the form value
```
Comment (markwhitfeld) [inline docs/plugins/form.md]: Missed one tweak
```suggestion
- `UpdateFormValue({ value, path, propertyPath? })` - Update the form value (or optionally an inner property value)
```

File: docs/plugins/form.md [hunk 8]
```diff
@@ -127,9 +127,88 @@ this.store.dispatch(
...
 - `UpdateFormDirty({ dirty, path })` - Update the form dirty status
 - `SetFormDisabled(path)` - Set the form to disabled
 - `SetFormEnabled(path)` - Set the form to enabled
 - `SetFormDirty(path)` - Set the form to dirty (shortcut for `UpdateFormDirty`)
 - `SetFormPristine(path)` - Set the form to pristine (shortcut for `UpdateFormDirty`)
+
+### Updating Form Properties Separately
+
+The form plugin exposes `UpdateFormValue` action that gives the opportunity to update nested form properties by providing `propertyPath` parameter.
```
Comment (markwhitfeld) [inline docs/plugins/form.md]:
```suggestion
The form plugin exposes the `UpdateFormValue` action that provides the ability to update nested form properties by supplying a `propertyPath` parameter.
```

File: docs/plugins/form.md [hunk 9]
```diff
@@ -127,9 +127,88 @@ this.store.dispatch(
...
 - `UpdateFormStatus({ status, path })` - Update the form status
-- `UpdateFormValue({ value, path })` - Update the form value
+- `UpdateFormValue({ value, path, propertyPath? })` - Update the form value
 - `UpdateFormDirty({ dirty, path })` - Update the form dirty status
 - `SetFormDisabled(path)` - Set the form to disabled
 - `SetFormEnabled(path)` - Set the form to enabled
 - `SetFormDirty(path)` - Set the form to dirty (shortcut for `UpdateFormDirty`)
 - `SetFormPristine(path)` - Set the form to pristine (shortcut for `UpdateFormDirty`)
+
+### Updating Form Properties Separately
```
Comment (markwhitfeld) [inline docs/plugins/form.md]:
```suggestion
### Updating Specific Form Properties
```

File: docs/plugins/form.md (hunk 1 above)
Comment (markwhitfeld) [reply docs/plugins/form.md]: Didn't quite get all of this.

**Comment by arturovt** [comment]
@markwhitfeld so we've release 3.5.1, can this be merged?

**Pull request merged**